### PR TITLE
Scripts/Temple of Ahn'Qiraj: Skeram spell cast update

### PR DESCRIPTION
**Pull request by ghost**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

-  After hit target 785 should cast on hit target spell id 2313.
-  
-  

**Target branch(es):** 3.3.5/master
- [X] 3.3.5
- [X] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)
Does it build

**Known issues and TODO list:** (add/remove lines as needed)
- [ ] 
- [ ] 

[//]: # (**DEPRECATION NOTICE** Instead of creating new PRs to the (old) 6.x branch, make sure the target branch is **master** instead of 6.x.)
[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request (if not enabled already).)
[//]: # (**NOTE** If this PR __only__ contains SQL files, open a new issue instead and post or link the SQL there.)
[//]: # (**SUGGESTION** When adding new SQL files, name them 9999_99_99_99_db_name.sql to reduce the chance of possible merge conflicts.)

### CODE REVIEW

File: src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp [hunk 1]
```diff
@@ -84,6 +85,12 @@ class boss_skeram : public CreatureScript
                     ((TempSummon*)me)->UnSummon();
             }
 
+            void SpellHitTarget(Unit* target, SpellInfo const* spell) override
+            {
+                if (spell->Id == SPELL_TRUE_FULFILLMENT)
+                    target->CastSpell(target, SPELL_TRUE_FULFILLMENT_2, true, NULL, NULL, me->GetGUID());
```
Comment (Keader) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp]: change ```NULL``` for ```nullptr```
btw, it cannot be: ```target->CastSpell(target, SPELL_TRUE_FULFILLMENT_2, true)``` ?

Comment (joschiwald) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp]: caster should not equal target (spell has TARGET_UNIT_TARGET_ANY)

Comment (Ulduar) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp]: nullptr - optional change - you can do without it.
caster should not equal target - why? What's the difference?

@Unknown-1000 - you can close and solve problem with sql part.
``` SQL
DELETE FROM `spell_linked_spell` WHERE `spell_trigger` = 784;
INSERT INTO `spell_linked_spell` (`spell_trigger`, `spell_effect`, `type`, `comment`) VALUES
(784,2313,1,'True Fulfillment Charm - Cast True Fulfillment Resistance part.');
```

Comment (Aokromes) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp]: spell_linked_spell is considerated hack.

Comment (Ulduar) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp]: Spellhit target also is considerated hack.

Comment (Keader) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp]: Trinity standards tells to use nullptr instead null :)
so please, always use standards :)

Comment (Ulduar) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp]: This is not my pull request. Cannot change anything.
SpellScript - example?

File: src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp [hunk 2]
```diff
@@ -84,6 +85,12 @@ class boss_skeram : public CreatureScript
                     ((TempSummon*)me)->UnSummon();
             }
 
+            void SpellHitTarget(Unit* target, SpellInfo const* spell) override
+            {
+                if (spell->Id == SPELL_TRUE_FULFILLMENT)
+                    caster->CastSpell(target, SPELL_TRUE_FULFILLMENT_2, true);
```
Comment (Keader) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp]: where this "caster" come from?

**Comment by Unknown-1000** [comment]
Enough! This is now.
Spell hit target = hack.
Spell linked = hack.
Spell script = hack.
Use spell linked. Reload and all. And easier.
Close request.

**Comment by Shauren** [comment]
HAHAHA, you won't get far with an attitude like that. For a fix to be accepted it must both work and be solved in a proper way, "just anything that works" wont pass here.

**Comment by Keader** [comment]
if you can't handle with feedbacks, you dont should open PRs :(

**Comment by offl** [comment]
Looks like that was the last straw for him :/

**Comment by Ulduar** [comment]
https://github.com/TrinityCore/TrinityCore/blob/5411ca8cdfc1f887db765f1d594c109e95628c54/src/server/game/Spells/Auras/SpellAuras.cpp#L1279

``` CPP
case 179415: // Crashing Tide
                        if (target->GetTypeId() == TYPEID_UNIT)
                            target->GetMotionMaster()->MoveRotate(10000, ROTATE_DIRECTION_RIGTH);
```
For example.

For Skeram case:
```CPP
case 785: // True Fulfillment Mind Control
       caster->CastSpell(target, 2313, true); // True Fulfillment resistance)
```

This part - not hacking. I hope, this is not wrong.

**Comment by Keader** [comment]
@Ulduar try with #18486 :)

**Comment by Ulduar** [comment]
SpellScript
For SPELL_AURA_MOD_CHARM?

Besides already added this option. And it works.
``` CPP
case 785: // True Fulfillment Mind Control
       caster->CastSpell(target, 2313, true); // True Fulfillment resistance)
```

**Comment by tkrokli** [comment]
@Ulduar : notice that this Pull Request is CLOSED and that Keader opened a new one in #18486 .
I recommend that you post your suggestions there, if you think Keader's PR could be improved.

**Comment by Ulduar** [comment]
Trom'ka @tkrokli 
I know this, don't worry  @Keader received this message.

**Pull request closed without merging**